.github/workflows/main.yml

```diff
@@ -5,30 +5,59 @@ on: [push, pull_request, workflow_dispatch]
 
 jobs:
   run_test:
-    name: Test on python ${{ matrix.python-version }}
+    name: ${{ matrix.py-ver-mypy-protobuf }} / ${{matrix.py-ver-mypy-target}} / ${{matrix.py-ver-unit-tests}} protoc/mypytarget/unittests py versions
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [2.7, 3.5, 3.6, 3.7, 3.8]
+        # Running mypy-protobuf itself
+        py-ver-mypy-protobuf: [3.8.6]
+        # --python-version=X target passed into mypy
+        py-ver-mypy-target: [3.5, 2.7]
```

Comment:
I think you dont need this. It is complicated to combine iterative matrix with additional include list. Using just include list would be much clear. (also you have only 2 alternatives accordingly to the matrix)

Reply:
It was motivated by a desire to increase this to include 3.6, 3.7, 3.8, 3.9 - so it would become the big one soon.

Perhaps more clear would be two separate build plans - one intending to test many targets holding the rest constant.

One intending to test many py-ver-mypy-protobuf holding the mypy-target constant.

That requires more refactoring - I'll punt on it for now.